compose/compose.go

```diff
@@ -572,6 +575,85 @@ func fatal(err error) {
 	os.Exit(1)
 }
 
+func makeDir(path string) error {
+	_, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		if errs := os.MkdirAll(path, 0755); errs != nil {
+			fatal(errors.Errorf("Couldn't create directory %v. Error: %v", path, errs))
+		}
+	} else if err != nil {
+		fatal(errors.Errorf("Something went wrong while checking if directory %v still exists. Error: %v",
+			path, err))
+	}
+	return nil
+}
+
+func copyFile(src, dst string) error {
+	var err error
+	var srcfd *os.File
+	var dstfd *os.File
+	var srcinfo os.FileInfo
+
+	if srcfd, err = os.Open(src); err != nil {
+		fatal(errors.Errorf("Error in opening source file %v. Error: %v", src, err))
+		return err
```

Comment:
Will this function return an error? Looks like it calls `fatal`, which will exit the program, right? IMO, best to just return the error and have it handled by the code that is calling copyFile

Reply:
Makes sense. Made function return an error.